### PR TITLE
openstack/nfv: find PCI address using sriovnetworknodestates

### PR DESCRIPTION
**Pull request by EmilienM**
Instead of hardcoding the PCI address, let's just get the real one
attached to the desired network; so we improve compatibility with
how machines are configured (q35, etc).

### CODE REVIEW

**Comment by EmilienM** [comment]
The job would randomly fail because we need https://github.com/openshift/sriov-network-operator/pull/837 but let's see how it works (and also to test the newly deployed NFV cloud on OSP 17.1):

/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.15-e2e-openstack-nfv-intel

**Comment by EmilienM** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-e2e-openstack-nfv-intel

File: ci-operator/step-registry/openstack/test/dpdk/openstack-test-dpdk-commands.sh [hunk 1]
```diff
@@ -168,7 +171,7 @@ else
     exit 1
 fi
 
-TESTPMD_OUTPUT=$(oc -n "${CNF_NAMESPACE}" rsh "${CNF_POD}" bash -c "yes | testpmd -l 2-3 --in-memory --allow 00:04.0 --socket-mem 1024 -n 4 --proc-type auto --file-prefix pg  -- --disable-rss  --nb-cores=1 --rxq=1 --txq=1 --auto-start --forward-mode=mac")
+TESTPMD_OUTPUT=$(oc -n "${CNF_NAMESPACE}" rsh "${CNF_POD}" bash -c "yes | testpmd -l 2-3 --in-memory --allow ${DPDK_PCI_DEVICE} --socket-mem 1024 -n 4 --proc-type auto --file-prefix pg  -- --disable-rss  --nb-cores=1 --rxq=1 --txq=1 --auto-start --forward-mode=mac")
```
Comment (EmilienM) [inline ci-operator/step-registry/openstack/test/dpdk/openstack-test-dpdk-commands.sh:174]: this one I'm not sure if it'll be happy. @zivgreenberg if you can comment on this one. Thanks

Comment (EmilienM) [reply ci-operator/step-registry/openstack/test/dpdk/openstack-test-dpdk-commands.sh:174]: in other words, do I need to trim the PCI device to remove the first two 00 ?

**Comment by EmilienM** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-e2e-openstack-nfv-intel

File: ci-operator/step-registry/openstack/test/dpdk/openstack-test-dpdk-commands.sh (hunk 1 above)
Comment (zivgreenberg) [reply ci-operator/step-registry/openstack/test/dpdk/openstack-test-dpdk-commands.sh:174]: @EmilienM There is no need to trim. It should work fine as long as the bus address is correct.

**Comment by EmilienM** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-e2e-openstack-nfv-hwoffload

**Comment by EmilienM** [comment]
`openstack-nfv-intel` is broken until we fix the hardware and merge https://github.com/openshift/sriov-network-operator/pull/837 so for this PR we'll rely on `openstack-nfv-hwoffload`.

/hold cancel

@mandre @MaysaMacedo if you see `openstack-nfv-hwoffload` rehearse passing, we can pj-rehearse ack and lgtm (if the change looks good to you); so we'll be able to test https://github.com/openshift/sriov-network-operator/pull/837 as soon as the HW is fixed (see VEXXHOST ticket).

Thanks

**Comment by mandre** [comment]
The [openstack-test-sriov step](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/43986/rehearse-43986-periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-e2e-openstack-nfv-hwoffload/1709759326962323456/artifacts/e2e-openstack-nfv-hwoffload/openstack-test-sriov/build-log.txt) failed with
```
Cloud openstack was not found.
```

Looks like we need to tell it where to find the `clouds.yaml` file. I'll update the patch.

**Comment by mandre** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-e2e-openstack-nfv-hwoffload

**Review by mandre — COMMENTED** [review]
/lgtm
/approve
/pj-rehearse ack

**Pull request merged**